src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadonlyStructTests.cs

```diff
@@ -1462,6 +1825,57 @@ .maxstack  2
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void ReadOnlyGetAccessor_CallReadOnlyGetAccessor()
+        {
+            var csharp = @"
+public struct S
+{
+    public int i;
+    public readonly int P1 { get => P2 + 1; }
+    public readonly int P2 { get => i; }
+}
+";
+            var comp = CompileAndVerify(csharp);
+            comp.VerifyIL("S.P1.get", @"
+{
+  // Code size        9 (0x9)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  call       ""int S.P2.get""
+  IL_0006:  ldc.i4.1
+  IL_0007:  add
+  IL_0008:  ret
+}
+");
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void ReadOnlyGetAccessor_CallAutoGetAccessor()
+        {
+            var csharp = @"
+public struct S
+{
+    public readonly int P1 { get => P2 + 1; }
+    public int P2 { get; }
+}
+";
+            var comp = CompileAndVerify(csharp);
+            comp.VerifyIL("S.P1.get", @"
+{
+  // Code size        9 (0x9)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  call       ""int S.P2.get""
+  IL_0006:  ldc.i4.1
+  IL_0007:  add
+  IL_0008:  ret
+}
+");
+            comp.VerifyDiagnostics();
+        }
```

Comment:
> } [](start = 7, length = 2)

I'll collect some test ideas here:
- Can you add a test for `readonly static` property or method? (expecting error)

Reply:
- `readonly static` in source is tested in ReadOnlyStructsTests. Do you want me to test what happens when a static method in metadata has an `IsReadOnlyAttribute`?
- I believe `readonly` local functions aren't possible. Because `this` is a ref local in a struct instance method, it can't be referenced in a local function. However I agree that this should be covered.

Reply:
I can't see the `ReadOnlyStructsTests` tests in this PR. Consider verifying `IsDeclaredReadOnly` and `IsEffectivelyReadOnly` in that scenario.

The scenario you mentioned (bad metadata with both `static` and `IsReadOnlyAttribute`) is also a good idea.

---
In reply to: [267508931](https://github.com/dotnet/roslyn/pull/34260#discussion_r267508931) [](ancestors = 267508931)

Reply:
Done.

Reply:
> I can't see the ReadOnlyStructsTests tests in this PR. Consider verifying IsDeclaredReadOnly and IsEffectivelyReadOnly in that scenario.

Consider adding covering that in a follow-up PR. (I don't think it was addressed in the current PR)

---
In reply to: [267524537](https://github.com/dotnet/roslyn/pull/34260#discussion_r267524537) [](ancestors = 267524537,267508931)